core/src/main/scala/cats/data/NonEmptySeq.scala

```diff
@@ -260,6 +260,23 @@ final class NonEmptySeq[+A] private (val toSeq: Seq[A]) extends AnyVal with NonE
   def reverse: NonEmptySeq[A] =
     new NonEmptySeq(toSeq.reverse)
 
+  def mapAccumulate[S, B](init: S)(f: (S, A) => (S, B)): (S, NonEmptySeq[B]) = {
+    var state = init
+
+    val fb = iterator.map { a =>
+      val (newState, b) = f(state, a)
+
+      state = newState
+
+      b
+    }.toIndexedSeq
+
+    (state, new NonEmptySeq(fb))
+  }
```

Comment:
Is that `.toIndexedSeq` still 100% safe here? I mean, `IndexedSeq` is an interface. Although all `IndexedSeq`-s from the Scala library might be strict collections, but we cannot completely exclude a case when someone can try to create a `NonEmptySeq` out of their own `IndexedSeq` implementation that have `iterator.toIndexedSeq` returning a non-strict collection. E.g. imagine some storage-efficient sparse-array implementation. If it happens then `state` won't be properly calculated.

I'd suggest to iterate the iterator directly via `hasNext`/`next` methods – that will definitely guarantee the enforced evaluation. Also it will probably make the implementation even more efficient since there will be no closure involved anymore.

Reply:
Actually, there is one in the stdlib `IndexedSeqView` which means yeah this is unsafe.